dask/array/tests/test_random.py

```diff
@@ -161,6 +161,11 @@ def test_random_all():
     da.random.standard_normal(size=5, chunks=3).compute()
     da.random.standard_t(2, size=5, chunks=3).compute()
 
+def test_array_broadcasting():
+    arr = np.arange(6).reshape((2,3))
+    assert da.random.poisson(arr, chunks=3).compute().shape == (2,3)
+    assert da.random.normal(arr, 2, chunks=3).compute().shape == (2,3)
```

Comment:
Thoughts on broadcasting or mix-and-matching?

```python
In [1]: import numpy as np

In [2]: np.random.normal(np.ones((2, 1)), np.ones((1, 4)))
Out[2]: 
array([[ 1.16711089, -1.6465026 ,  0.28312835,  1.02837577],
       [ 0.55505135,  0.24445202,  0.99830937,  1.44645849]])

In [3]: np.random.normal(np.ones((2, 1)), 10, size=(2, 4))
Out[3]: 
array([[  0.89357538,  -3.29337928,   2.81504948, -13.1528982 ],
       [  3.72773089,  -7.31671413,  15.83772526,  -2.89272702]])
```

It's also fine to raise `NotImplementedError` on any of these if we don't want to deal with supporting them, but we need to make sure that if we expose this functionality that we always return sane responses.